### DISCUSSION TITLE
Having trouble building

### DISCUSSION THREAD

**Discussion by KarimElsayad247** [H-M-H/Weylus#311] [Q&A] [answered]
tldlr: it doesn't find ffmpeg headers

```
warning: weylus@0.11.4: lib/encode_video.c:5:10: fatal error: libavcodec/avcodec.h: No such file or directory
warning: weylus@0.11.4:     5 | #include <libavcodec/avcodec.h>
warning: weylus@0.11.4:       |          ^~~~~~~~~~~~~~~~~~~~~~
warning: weylus@0.11.4: compilation terminated.
```

I build with `cargo build --features ffmpeg-system`

looking at the system include dir

```
$ ls /usr/include/ffmpeg
libavcodec  libavutil  libswresample
```
```
$ ls /usr/include/ffmpeg/libavcodec/
ac3_parser.h   avcodec.h  avfft.h  codec_desc.h  codec_id.h   d3d11va.h  dirac.h       dxva2.h  mediacodec.h  qsv.h    version.h        videotoolbox.h
adts_parser.h  avdct.h    bsf.h    codec.h       codec_par.h  defs.h     dv_profile.h  jni.h    packet.h      vdpau.h  version_major.h  vorbis_parser.h
```

So the dependencies are already there... I didn't have any trouble building on Ubuntu 24.04, but this problem arose when I tried building on Opensuse Tumbleweed

I tried modifying the `build.rs` file and pass the `-I` flag to `let mut cc_video = cc::Build::new();` like so

```
    if target_os == "linux" {
        cc_video.define("HAS_VAAPI", None)
                  .flag("-I /usr/include/ffmpeg"); // added this
    }
```

but to no avail.

Any ideas?

**Comment by KarimElsayad247** [comment, marked as answer]
These two problems seem relevant 

https://github.com/tony2001/ffmpeg-php/pull/21

https://github.com/tony2001/ffmpeg-php/issues/24

And the solution is to copy the headers directly into `include` dir

```
https://github.com/tony2001/ffmpeg-php/issues/24
```

**Reply by H-M-H** [reply]
You can also try CC's `include` https://docs.rs/cc/latest/cc/struct.Build.html#method.include or add `/usr/include/ffmpeg` to the `C_INCLUDE_PATH` env var.
